### ISSUE TITLE
new tests in parallel e2e run can fail due to resource contention

### ISSUE THREAD

**Issue by kargakis** [kubernetes/kubernetes#25140]
```
â€¢ Failure [357.058 seconds]
[k8s.io] Kubectl client
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:502
  [k8s.io] Kubectl rolling-update
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:502
    should support rolling-update to same image [Conformance] [It]
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:1062

    May  4 08:14:22.245: Timed out after 300 seconds waiting for run=e2e-test-nginx-rc pods to reach valid state

    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:1458
```

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/23590/kubernetes-pull-build-test-e2e-gce/38280/?pli=1

@jlowdermilk @kubernetes/kubectl

**Comment by jlowdermilk** [comment]
After moving this test to the flaky suite, it's failed [once](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-flaky/16340) in ~200 runs, and that was because a kubelet ran out of disk space. The test itself is not flaky, so what likely happened is that adding it to the parallel suite causes resource contention.

@spxtr, is there a precedent for increasing the size (either node size or num nodes) of the cluster launched for parallel runs? Or alternatively, can we decrement the number of tests ginkgo is allowed run in parallel to deal with resource contention?

**Comment by spxtr** [comment]
This came up recently in #20461. We can consider bumping up to 4 nodes for parallel tests, or using a smaller number of ginkgo nodes than [30](https://github.com/kubernetes/kubernetes/blob/master/hack/ginkgo-e2e.sh#L81).

**Comment by jlowdermilk** [comment]
This should be addressed (for now) by #25548. We'll eventually need another node in test clusters, but lowering parallelism is reasonable bandaid.

**Comment by ixdy** [comment]
Do we have a way to monitor test cluster usage so we can address these issues proactively?
